### PR TITLE
#385 dont hide declined event  in the main calendar

### PR DESCRIPTION
**Pull request by asmaNode**
https://user-images.githubusercontent.com/43316837/105485631-80817f00-5cad-11eb-9b18-baf1eeb90234.mp4

### CODE REVIEW

File: src/esn.calendar.libs/app/services/cached-event-source.spec.js [hunk 1]
```diff
@@ -747,8 +750,25 @@ describe('The calCachedEventSource service', function() {
         self.CAL_ICAL.partstat.tentative
       ].forEach(expectedEvents);
     });
+    it('should return all event when user he is not the owner of calendar', function() {
```
Comment (huy-ta) [inline src/esn.calendar.libs/app/services/cached-event-source.spec.js:753]:
```suggestion

    it('should return all event when user he is not the owner of calendar', function() {
```

File: src/esn.calendar.libs/app/services/cached-event-source.spec.js [hunk 2]
```diff
@@ -747,8 +750,25 @@ describe('The calCachedEventSource service', function() {
...
+        attendees: []
+      };
+      var anEvent = newEvent(attrs);
+
+      self.eventsMock.push(anEvent);
+
+      getWrapEventSource(self.eventsMock);
+      self.$rootScope.$digest();
+
+      expect(self.originalCallback).to.have.been.calledWith(self.eventsMock);
```
Comment (rezk2ll) [inline src/esn.calendar.libs/app/services/cached-event-source.spec.js:767]: needs to expect that the callback returned with `true`

File: src/esn.calendar.libs/app/services/cached-event-source.js [hunk 3]
```diff
@@ -197,15 +197,18 @@ require('./cal-ui-authorization-service.js');
...
       if (!calFullUiConfiguration.isDeclinedEventsHidden()) {
         return events;
       }
 
       return events.filter(function(event) {
         var userAsAttendee = calEventUtils.getUserAttendee(event);
 
-        if (!userAsAttendee) {
+        if (!userAsAttendee || !calendarOwner) {
+
```
Comment (huy-ta) [inline src/esn.calendar.libs/app/services/cached-event-source.js:211]:
```suggestion
```

File: src/esn.calendar.libs/app/services/cached-event-source.js [hunk 4]
```diff
@@ -197,15 +197,17 @@ require('./cal-ui-authorization-service.js');
       });
     }
 
-    function _handleDeclinedEvents(events) {
+    function _handleDeclinedEvents(calendar, events) {
+      const calendarOwner = calendar.isOwner(session.user._id);
```
Comment (renaudboyer) [inline src/esn.calendar.libs/app/services/cached-event-source.js:201]: Does isOwner works when user A have has been delegated on user B calendar and user B have declined an event ?

Comment (asmaNode) [reply src/esn.calendar.libs/app/services/cached-event-source.js:201]: yes it is working , will put a demo with this senario

Comment (asmaNode) [reply src/esn.calendar.libs/app/services/cached-event-source.js:201]: **Demo with delegation:**
https://user-images.githubusercontent.com/43316837/105775518-362b2700-5f67-11eb-802e-59634270c5fe.mp4

**Review by rezk2ll — DISMISSED** [review]

**Review by rezk2ll — APPROVED** [review]

**Review by renaudboyer — APPROVED** [review]

**Review by huy-ta — APPROVED** [review]

**Pull request merged**